Modules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx

```diff
@@ -280,29 +273,40 @@ template <typename TInputImage>
 void
 ContourExtractor2DImageFilter<TInputImage>::GenerateDataForLabels()
 {
-  using IndexType = typename InputRegionType::IndexType;
-  using SizeType = typename InputRegionType::SizeType;
-  // using RegionRange = ImageRegionRange<InputImageType>;
-  using RegionConstRange = ImageRegionRange<const InputImageType>;
-  // using RegionIterator = ImageRegionIterator<InputImageType>;
-  using RegionConstIterator = ImageRegionConstIterator<InputImageType>;
-
-  // Requested region
-  const InputRegionType  inputRegion{ this->GetInput()->GetRequestedRegion() };
-  const RegionConstRange inputRange{ *this->GetInput(), inputRegion };
+  // We want to make use of ConstantBoundaryCondition, which works
+  // with *images*, but use it with the input *region*.  So, we copy
+  // the region to be its own image if necessary.
+  InputImagePointer      inputGC{ nullptr };
+  const InputImageType * input{ this->GetInput() };
+  const InputRegionType  inputRegion{ input->GetRequestedRegion() };
+  // We can remove this if statement and its contents once a Policy (e.g.,
+  // itk::ConstantBoundaryImageNeighborhoodPixelAccessPolicy<InputImageType>) can support constant values outside of the
+  // region (rather than merely outside of the image).
+  if (input->GetLargestPossibleRegion() != inputRegion)
+  {
+    inputGC = InputImageType::New();
+    inputGC->SetRegions(inputRegion);
+    inputGC->Allocate();
+    const RegionRange      inputRange{ *inputGC, inputRegion };
+    const RegionConstRange originalRange{ *this->GetInput(), inputRegion };
+    std::copy(originalRange.cbegin(), originalRange.cend(), inputRange.begin());
```

Comment:
I thought we would have to make a copy for each label's region. Just one copy is not too bad. Of course, solving this via boundary condition is preferred. But that can be left for a follow-up PR.

Reply:
Yes, just one copy overall ... because the problem case that the extended region addresses is when the label of interest is also present just outside the region of interest.  Since the label-specific bounding boxes guarantee that the label of interest won't be inside the region but outside the bounding box, we don't need to special-case the bounding box boundary.

Yes, a follow-up PR when we have a boundary condition for regions is what I am thinking too.